plugin/trino-raptor-legacy/src/test/java/io/trino/plugin/raptor/legacy/metadata/TestDatabaseShardManager.java

```diff
@@ -357,14 +346,12 @@ public void testExternalBatches()
 
         shards = ImmutableList.of(shardInfo(UUID.randomUUID(), "node1"));
 
-        try {
-            transactionId = shardManager.beginTransaction();
-            shardManager.commitShards(transactionId, tableId, columns, shards, externalBatchId, 0);
-            fail("expected external batch exception");
-        }
-        catch (TrinoException e) {
-            assertEquals(e.getErrorCode(), RAPTOR_EXTERNAL_BATCH_ALREADY_EXISTS.toErrorCode());
-        }
+        transactionId = shardManager.beginTransaction();
+        long finalTransactionId = transactionId;
+        List<ShardInfo> finalShards = shards;
+        assertTrinoExceptionThrownBy(() -> shardManager.commitShards(finalTransactionId, tableId, columns, finalShards, externalBatchId, 0))
+                .hasErrorCode(RAPTOR_EXTERNAL_BATCH_ALREADY_EXISTS)
+                .hasMessage("External batch already exists: foo");
     }
```

Comment:
same comment as above: shall we put transactionId = shardManager.beginTransaction(); inside lambda?

Reply:
@MariyaR I've fixed the code as suggested.